Offline stream and private stream player errors

clacladev opened the pull request:
## Description
Introduced the offline and private livestream player errors.

Fixes #237 


## Screenshots
<img width="743" alt="Screenshot 2022-12-19 at 13 26 08" src="https://user-images.githubusercontent.com/161903/208739324-ceefba4b-f42d-4c03-bafd-6a6766764980.png">
<img width="1512" alt="Screenshot 2022-12-19 at 13 26 35" src="https://user-images.githubusercontent.com/161903/208739329-3481e27d-723e-40e3-9190-ee0613f7f619.png">
<img width="870" alt="Screenshot 2022-12-20 at 17 10 17" src="https://user-images.githubusercontent.com/161903/208739331-fea23980-f39b-4886-ba48-3188a735b140.png">

Randy1Burrell commented:
It would probably be easier to apply this by changing the default value. @yondonfu what do you think?

yondonfu commented:
Seems reasonable (unless there is some UI reason for limiting the # of decimals)

Randy1Burrell commented:
There are some UI effects I saw when I changed the default value for decimals, but I have a fix for that.

0xcadams committed:
Add React Core Package (#141)

* docs: update nextra dependency

* feat: added crypto package and work on signed playback

* fix: update deps

* fix: types

* fix: move to using rainbowkit

* feat: removed dependencies and used webcrypto for jwt signing

* chore: add initial expo dev example app

* chore: added react native bare dev

* fix: further work towards rn examples

* Added z-index to the default variant of the player (#132)

* Add nodejs example (#131)

* feat: added import script for nodejs example

* fix: small fixes and added comments

* fix: added handling of paths in dstorage

* fix: moved ipfs-playback to dstorage and added redirect

* fix: added output json and fix docs links

* fix: fix docs on contributing

* chore(deps): bump pnpm/action-setup from 2.2.2 to 2.2.4 (#125)

Bumps [pnpm/action-setup](https://github.com/pnpm/action-setup) from 2.2.2 to 2.2.4.
- [Release notes](https://github.com/pnpm/action-setup/releases)
- [Commits](https://github.com/pnpm/action-setup/compare/v2.2.2...v2.2.4)

---
updated-dependencies:
- dependency-name: pnpm/action-setup
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Chase Adams <c@cadams.io>

* Added changeset for z-index change (#135)

* fix: added HEAD request to check for valid URLs

* Add access control docs (#93)

* docs: update nextra dependency

* feat: added crypto package and work on signed playback

* fix: update deps

* fix: types

* fix: move to using rainbowkit

* feat: removed dependencies and used webcrypto for jwt signing

* feat: added further access control

* fix: added b64 decoding for private key

* fix: final fixes for access control

* fix: fixed packages

* docs: clean up svelte example

* chore: fix up issue templates

* docs: removed extra brackets from useCreateAsset example (#136)

Co-authored-by: Chase Adams <c@cadams.io>

* update docs (#138)

* update docs

* Update docs/pages/examples/react/dstorage-playback.en-US.mdx

Co-authored-by: Chase Adams <c@cadams.io>

Co-authored-by: Chase Adams <c@cadams.io>

* chore: version packages (#134)

* chore: version packages

* ci: bump

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Chase Adams <c@cadams.io>

* Add support for multiple files at once using useCreateAsset  (#102)

* feat: added multi-video upload to useCreateAsset hook

* fix: updated changeset description

* fix: solved the Asset.tsx in example component error

* feat: added multiple file upload at the core package

* fix: fixed the build error in next example directory

* fix: refactor types

* fix: updated the changeset and types

* docs: updated the types in useCreateAsset page

* fix: fixed the merge conflicts

* docs: updated the types in the documentation.

* refactor: removed styles and console.log's

* refactor: replaced promise.all with promise.allSettled

* docs: updated the types in the docs

* refactor: updated the upload progress object

* fix: reverted the types

- refactor: updated the return in Promise.allSettled

* fix: fixed the typed lint issue

* refactor: remove the uploadsize from tus

* fix: fix types

* refactor: removed meta from the tus upload

* fix: refactored types

* docs: added new types to the docs

Co-authored-by: Chase Adams <c@cadams.io>

* Rework types to support multiple createAsset as const (#140)

* feat: added multi-video upload to useCreateAsset hook

* fix: updated changeset description

* fix: solved the Asset.tsx in example component error

* feat: added multiple file upload at the core package

* fix: fixed the build error in next example directory

* fix: refactor types

* fix: updated the changeset and types

* docs: updated the types in useCreateAsset page

* fix: fixed the merge conflicts

* docs: updated the types in the documentation.

* refactor: removed styles and console.log's

* refactor: replaced promise.all with promise.allSettled

* docs: updated the types in the docs

* refactor: updated the upload progress object

* fix: reverted the types

- refactor: updated the return in Promise.allSettled

* fix: fixed the typed lint issue

* refactor: remove the uploadsize from tus

* fix: fix types

* refactor: removed meta from the tus upload

* fix: refactored types

* docs: added new types to the docs

* fix: rework types to support const assertions for typing

* test: fix tests

Co-authored-by: Suhail Kakar <suhail.zahed@gmail.com>

* fix: merge cleanup

* test: fix tests

* fix: update nodejs example

* refactor: added core-react package and moved web logic out of react

* chore: changesets

* Revert "chore: changesets"

This reverts commit 498cfca9d1ff56a5dc3a9cfcfb702ed3ec8e1569.

* chore: changeset

* ci: add release on next branch

* chore: add pre

* fix: fix prerendering

* style: lint

* fix: added core js to react

* test: fix tests

* test: fix

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: clacla <claudio@livepeer.org>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Suhail Kakar <suhail.zahed@gmail.com>
Co-authored-by: pglowacky <79172645+pglowacky@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

0xcadams committed:
Merge remote-tracking branch 'origin/main' into next

0xcadams committed:
test: fix tests

0xcadams committed:
Merge commit '661170013bc8a8ad485e3c9bde5cd0d20a97a145' into next

0xcadams committed:
Merge remote-tracking branch 'origin/main' into next

0xcadams committed:
Add Expo & Bare Dev Projects (#152)

* feat: added initial react native package

* feat: further work on monorepo

* chore: added react native example projects

* chore: small fixes

* fix: fixed types

* chore: move to local package

* test: testing removing node linker

* ci: attempt to fix ci

* fix: fix builds

* test: fix tests and remove rn tests due to lack of vite support :(

* test: attempt to fix tests

* ci: fix react 17 test

* test: fix tests to sort imports

0xcadams committed:
Merge branch 'main' into next

0xcadams committed:
Merge remote-tracking branch 'origin/main' into next

0xcadams committed:
ci: remove dep review on next branch

0xcadams committed:
[WIP] Add RN Controls (#157)

* feat: added sketch of buttons with stitches native

* fix: fixed multiple uploads (#154)

* chore: version packages (#156)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

* feat: added all remaining controls and broadcast package

* fix: reverted file fingerprint (#158)

* chore: version packages (#159)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

* refactor: refactored to add usePlayButton, ... hooks to core

* refactor: reworked theme to have shared types

* refactor: further cleanup

* fix: further refactoring

* fix: further refactoring

* fix: final fixes for build

* fix: final fixes from testing android/ios w/ expo/rn bare

* fix: fixed metrics to work in RN

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

0xcadams committed:
fix: fixed to use w3slink

0xcadams committed:
Merge remote-tracking branch 'origin/main' into next

0xcadams committed:
Merge branch '0xcadams/ipfs-cleanup' into next

0xcadams committed:
Add docs to RN (#168)

* fix: fixes for theming and added poster functionality

* docs: updated docs for react native

* fix: added back ethers dep

* fix: fix build

0xcadams committed:
fix: added memo to Player

0xcadams committed:
Merge commit '6fb4fde53d1f2363e7ee7f8c820c80a713018c0e' into next

0xcadams committed:
chore: changesets

0xcadams committed:
chore: update deps

github-actions[bot] committed:
chore: version packages (next) (#142)

* chore: version packages (next)

* ci: bump

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Chase Adams <c@cadams.io>

0xcadams committed:
Remove default api key (#172)

* feat: remove default api key

* docs: added sunset date

* test: fix tests

* test: fix tests

github-actions[bot] committed:
chore: version packages (next) (#173)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

0xcadams committed:
Merge remote-tracking branch 'origin/main' into next

0xcadams committed:
fix: add forwardRef

0xcadams committed:
fix: changed to use custom ref prop

0xcadams committed:
fix: added docs and moved ref to custom prop

0xcadams committed:
fix: fixed types

github-actions[bot] committed:
chore: version packages (next) (#177)

* chore: version packages (next)

* ci: bump

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Chase Adams <c@cadams.io>

0xcadams committed:
fix: publish core react

0xcadams committed:
fix: publish core react

github-actions[bot] committed:
chore: version packages (next) (#178)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

0xcadams committed:
chore: update

0xcadams committed:
Merge remote-tracking branch 'origin/next' into next

0xcadams committed:
fix: added core react

github-actions[bot] committed:
chore: version packages (next) (#179)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

0xcadams committed:
fix: move to use callback refs

0xcadams committed:
Merge remote-tracking branch 'origin/next' into next

github-actions[bot] committed:
chore: version packages (next) (#180)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

0xcadams committed:
Refactor core web (#182)

* refactor: refactored web out of core and into parent module

* fix: add further and changeset

* fix: remove ethers

* test: fix tests

* test: fix and add more tests

* chore: update deps and fix tests

0xcadams committed:
chore: change changeset to major for new core

0xcadams committed:
docs: fix install command

github-actions[bot] committed:
chore: version packages (next) (#183)

* chore: version packages (next)

* ci: bump

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Chase Adams <c@cadams.io>

0xcadams committed:
docs: fix docs for nextra 2

0xcadams committed:
fix: slider bug (#184)

github-actions[bot] committed:
chore: version packages (next) (#185)

* chore: version packages (next)

* ci: bump

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Chase Adams <c@cadams.io>

0xcadams committed:
docs: fix docs based on feedback

0xcadams committed:
chore: add monitoring

0xcadams committed:
Add further theme tokens (#187)

* chore: added sentry

* feat: added further customization to sliders

* fix: remove console log

github-actions[bot] committed:
chore: version packages (next) (#188)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

0xcadams committed:
fix: fix regression (#189)

github-actions[bot] committed:
chore: version packages (next)

0xcadams committed:
ci: bump

github-actions[bot] committed:
chore: version packages (next) (#190)

* chore: version packages (next)

* ci: bump

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Chase Adams <c@cadams.io>

0xcadams committed:
fix: fix fallback

0xcadams committed:
chore: add changeset

0xcadams committed:
Merge branch 'changeset-release/next' into next

github-actions[bot] committed:
chore: version packages (next) (#191)

* chore: version packages (next)

* ci: bump

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Chase Adams <c@cadams.io>

0xcadams committed:
fix: fixed src autoplayback

github-actions[bot] committed:
chore: version packages (next) (#192)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

0xcadams committed:
feat: added showUploadingIndicator (#193)

0xcadams committed:
docs: added docs for showUploadingIndicator

github-actions[bot] committed:
chore: version packages (next) (#194)

* chore: version packages (next)

* ci: bump

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Chase Adams <c@cadams.io>

0xcadams committed:
docs: fixed line highlighting

0xcadams committed:
Add loading color (#195)

* feat: added loading color

* ci: bump

* fix: fixed missing theme config

github-actions[bot] committed:
chore: version packages (next) (#196)

* chore: version packages (next)

* ci: bump

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Chase Adams <c@cadams.io>

0xcadams committed:
fix: changed default to contain (#197)

github-actions[bot] committed:
chore: version packages (next) (#198)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

Commit:
Created the basic scaffolding to show the Offline Stream error

Commit:
Added the styled offline error

Commit:
Hooked the private stream error management

Commit:
Displaying the private stream error

Commit:
Extracted the svgs in separate components

Commit:
Checking the HLS error

Commit:
Merge branch 'main' of https://github.com/livepeer/livepeer.js into clacladev/player-errors

Commit:
Fixed the SVGs related warnings

Commit:
Fixed stitched theme on react-native

Commit:
Merge branch 'main' of https://github.com/livepeer/livepeer.js into clacladev/player-errors

Commit:
Refactored the svg components

Commit:
Simplified isStreamOffline

Commit:
Added unfinished error tracking in the Video element

Commit:
Exporting the type the ThemeConfig is depending on

Commit:
Merge branch 'main' of https://github.com/livepeer/livepeer.js into clacladev/player-errors

Commit:
Fixed test error

0xcadams committed:
fix: nits and changeset

Commit:
Refactored how the player manages the offline stream

0xcadams committed:
Merge branch 'main' into clacladev/player-errors-new-logic

0xcadams reviewed (approved):
This is awesome, works perfectly and even continues to poll when a stream goes offline. cc @victorges 

Could you add a changeset detailing the changes made in this PR? 🚀 

https://github.com/livepeer/livepeer.js/blob/main/.github/CONTRIBUTING.md#versioning

0xcadams commented:
Actually I just cherry picked my changeset commit from the old version so firing away!
